addons/sys_intercom/fnc_configIntercomStations.sqf

```diff
@@ -0,0 +1,52 @@
+/*
+ * Author: ACRE2Team
+ * Configures the initial intercom connectivity (disconnected/connected) for all allowed seats.
+ *
+ * Arguments:
+ * 0: Vehicle with intercom <OBJECT>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [vehicle player] call acre_sys_intercom_fnc_configIntercomStations
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+params ["_vehicle"];
+
+private _type = typeOf _vehicle;
+
+private _intercomStatus = [];
+private _intercomPos = [];
+
+private _allowedPositions = _vehicle getVariable[QGVAR(allowedPositions), []];
+private _initialConfiguration = +(_vehicle getVariable[QGVAR(connectByDefault), []]);
+
+{
+    {
+        _intercomPos pushBackUnique _x;
+    } forEach _x; // position
+} forEach _allowedPositions;
+
+{
+    // Make a hard copy of the array otherwise when modifying the array, all items in the parent array will be changed
```

Comment:
Arrays can be copied with the `+` operator see: https://community.bistudio.com/wiki/Array#Copying